### PR TITLE
[FIX] payment_mercado_pago: sanitarize tx reference in webhook url

### PR DESCRIPTION
**Pull request by maq-adhoc**
Description of the issue/feature this PR addresses: The webhook URLs used by MercadoPago are not sanitized and cause the payment to fail when the URL contains spaces or other non-ASCII characters.

Steps to reproduce the bug:
1. Instal the Argentine localization and configure "payment_mercado_pago" as the payment method (test or production).
2. Create an invoice and generate a payment link (the reference will be something like "FA-C 0001-000133444").
3. Initiate the payment process and it will generate an error in the last stage of payment (on the MercadoPago website).

- Mercadopago tests credentials TEST-4811719641145832-020115-c5caffc149634e3e9d72fde2b74544d8__LD_LA__-172837665
- Used payment methods: https://www.mercadopago.com.ar/developers/es/docs/checkout-api/additional-content/your-integrations/test/cards

- Credencial de test usada TEST-4811719641145832-020115-c5caffc149634e3e9d72fde2b74544d8__LD_LA__-172837665
- Metodo de pago usados: https://www.mercadopago.com.ar/developers/es/docs/checkout-api/additional-content/your-integrations/test/cards

Current behavior before PR:
Payment in MercadoPago cannot be confirmed when the reference contains spaces.

Desired behavior after PR is merged:
Payment in MercadoPago is confirmed even when the reference contains spaces.

Adhoc Ticket: 68591

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_mercado_pago/models/payment_transaction.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 from odoo.addons.payment_mercado_pago.const import TRANSACTION_STATUS_MAPPING
 from odoo.addons.payment_mercado_pago.controllers.main import MercadoPagoController
-
+import urllib.parse
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_transaction.py:13]: Please move the import right after `import pprint`.

File: addons/payment_mercado_pago/models/payment_transaction.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def _mercado_pago_prepare_preference_request_payload(self):
         """
         base_url = self.provider_id.get_base_url()
         return_url = urls.url_join(base_url, MercadoPagoController._return_url)
+        sanitarized_reference = urllib.parse.quote(self.reference)
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_transaction.py:58]:
```suggestion
        sanitized_reference = urllib.parse.quote(self.reference)
```

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello, that looks good to me :)

Could you just fix the failing test?

File: addons/payment_mercado_pago/models/payment_transaction.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import logging
 import pprint
 
+from urllib.parse import quote as url_quote
 from werkzeug import urls
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_transaction.py:7]: `urllib` is a standard lib so it can go in the first group.
```suggestion
import pprint
from urllib.parse import quote as url_quote

from werkzeug import urls
```

File: addons/payment_mercado_pago/models/payment_transaction.py [hunk 4]
```diff
@@ -11,7 +12,6 @@
 from odoo.addons.payment_mercado_pago.const import TRANSACTION_STATUS_MAPPING
 from odoo.addons.payment_mercado_pago.controllers.main import MercadoPagoController
 
-
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_transaction.py:14]: unnecessary diff

File: addons/payment_mercado_pago/models/payment_transaction.py [hunk 5]
```diff
@@ -55,10 +55,10 @@ def _mercado_pago_prepare_preference_request_payload(self):
         """
         base_url = self.provider_id.get_base_url()
         return_url = urls.url_join(base_url, MercadoPagoController._return_url)
+        sanitized_reference = url_quote(self.reference)
         webhook_url = urls.url_join(
-            base_url, f'{MercadoPagoController._webhook_url}/{self.reference}'
+            base_url, f'{MercadoPagoController._webhook_url}/{sanitized_reference}'
         )  # Append the reference to identify the transaction from the webhook notification data.
-
```
Comment (AntoineVDV) [inline addons/payment_mercado_pago/models/payment_transaction.py:61]: Here too.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: addons/payment_mercado_pago/models/payment_transaction.py (hunk 2 above)
Comment (maq-adhoc) [reply addons/payment_mercado_pago/models/payment_transaction.py]: its fixed

File: addons/payment_mercado_pago/models/payment_transaction.py (hunk 3 above)
Comment (maq-adhoc) [reply addons/payment_mercado_pago/models/payment_transaction.py]: fixed

File: addons/payment_mercado_pago/models/payment_transaction.py (hunk 4 above)
Comment (maq-adhoc) [reply addons/payment_mercado_pago/models/payment_transaction.py]: Done

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello, the current fix is good but there is still a test failing. See https://runbot.odoo.com/runbot/build/55856363
Could you adapt the test to your changes?

**Comment by maq-adhoc** [comment]
> Hello, the current fix is good but there is still a test failing. See https://runbot.odoo.com/runbot/build/55856363 Could you adapt the test to your changes?

i love test!!! Fixed

**Review by AntoineVDV — APPROVED** [review]
Thanks :)

@robodoo r+
@robodoo override=ci/security
Reason: encoding of the transaction reference which is not based on user-provided data
Link to ci/security build (16.0): https://github.com/odoo/odoo/pull/147960
